### DISCUSSION TITLE
How to find a table element by json field?

### DISCUSSION THREAD

**Discussion by GongFu666** [prisma/prisma#17201] [Q&A]
Sorry, noob question. My prisma schema looks like this:

```
model email {
  id          Int @id @default(autoincrement())
  data        Json @unique
  created_at  DateTime @default(now()) @db.Timestamptz(6)
  updated_at  DateTime @default(now()) @db.Timestamptz(6)
}
```

Now I would like to find an email given a json data object. Something like:

`const res = await prisma.email.findUnique(...)`

What goes inside the brackets? Thanks for your help!

**Comment by nurul3101** [comment]
Hello @GongFu666 👋 

From your use case, it looks like you want to match entire JSON object, You can so something like this (Assuming a PostgreSQL database):
```typescript
var json = { [{ name: 'Bob the dog' }, { name: 'Claudine the cat' }] }

const getUsers = await prisma.email.findUnique({
  where: {
    data: {
      equals: json,
    },
  },
})
```

This would return the email which matches the JSON structure. 

Here's documentation [reference](https://www.prisma.io/docs/concepts/components/prisma-client/working-with-fields/working-with-json-fields#filter-on-a-json-field), for filtering on JSON values.

**Reply by GongFu666** [reply]
Thanks for your reply,
but it is not working for me. I create the database records in a loop with

```
const emailCreateInput: Prisma.emailCreateInput = {
  data: data[index],
}
const createEmail = await prisma.email.create({
  data: emailCreateInput,
})
```

and try to find them with

```
const emailWhereUniqueInput: Prisma.emailWhereUniqueInput = {
  data: data[index],
}
const getEmails = await prisma.email.findUnique({
  where: {
    data: {
      equals: emailWhereUniqueInput,
    },
  },
})
```
however, `getEmails` is always `null`, even though the creation succeeds.

What am I doing wrong?

EDIT: It seems like the order of the properties of the json object that ends up in the database is different from the order given in `emailWhereUniqueInputs`

**Reply by nurul3101** [reply]
Do you get the records when the order of properties in the JSON object of `emailWhereUniqueInputs` is same as the JSON object stored in the database?

**Reply by GongFu666** [reply]
No :weary:

**Reply by GongFu666** [reply]
I resorted to adding the message id of the email, which is supposed to be unique, to the email model. Now I can find the emails like this

```
const messageId: string = jsonData[index]['Message-Id']
getEmail = await prismaC.email.findUnique({
          where: {
            message_id: messageId,
          },
        })
```

However, I still don't know why I can't find the emails by json...

**Reply by nurul3101** [reply]
Did you try [filtering by object property](https://www.prisma.io/docs/concepts/components/prisma-client/working-with-fields/working-with-json-fields#filter-on-object-property) in a JSON?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
